app/routes/company/components/CompanyDetail.tsx

```diff
@@ -23,224 +27,138 @@ type Props = {
-              }}
-            />
-          </div>
-        )}
-      </div>
-    );
+  if (!company) {
+    return <LoadingIndicator loading={loading} />;
   }
-}
 
-const CompanyDetail = (props: Props) => {
-  const {
-    company,
-    companyEvents,
-    joblistings,
-    fetchMoreEvents,
-    showFetchMoreEvents,
-    loggedIn,
-  } = props;
+  const sortedEvents = companyEvents.sort(
+    (a, b) => moment(b.startTime).unix() - moment(a.startTime).unix()
+  );
+  const upcomingEvents = sortedEvents.filter((event) =>
+    moment().isBefore(moment(event.startTime))
+  );
+  const oldEvents = sortedEvents.filter((event) =>
+    moment().isAfter(moment(event.startTime))
+  );
 
-  if (!company) {
-    return <LoadingIndicator loading />;
-  }
+  const companyInfo = [
+    {
+      text: company.website,
+      icon: 'globe-outline',
+      link: true,
+    },
+    {
+      text: company.address,
+      icon: 'location-outline',
+      link: false,
+    },
+    {
+      text: company.phone,
+      icon: 'call-outline',
+      link: false,
+    },
+    {
+      text: company.companyType,
+      icon: 'briefcase-outline',
+      link: false,
+    },
+  ];
 
-  const joblistingsList =
-    joblistings &&
-    joblistings.map((joblisting) => (
-      <tr key={joblisting.id}>
-        <td>
-          <Link to={`/joblistings/${joblisting.id}`}>{joblisting.title}</Link>
-        </td>
-        <td>{jobType(joblisting.jobType)}</td>
-        <td>
-          <Year joblisting={joblisting} />
-        </td>
-        <td>
-          {
-            // TODO: this doesn't work, because applicationUrl doesn't exist with list-serializer
-            joblisting.applicationUrl && (
-              <a
-                href={joblisting.applicationUrl}
-                target="_blank"
-                rel="noreferrer"
-              >
-                <strong>SØK HER</strong>
-              </a>
-            )
-          }
-        </td>
-      </tr>
-    ));
   return (
-    <Content>
+    <Content
+      banner={company.logo || company?.logo}
+      bannerPlaceholder={company.logoPlaceholder || company?.logoPlaceholder}
+    >
       <Helmet title={company.name} />
-      {company.logo && (
-        <div>
-          <Image
-            src={company.logo}
-            className={styles.companyImage}
-            alt={`${company.name} logo`}
-          />
-        </div>
-      )}
-
       <NavigationTab
         title={company.name}
         back={{
-          label: 'Tilbake til liste',
+          label: 'Bedriftsoversikt',
           path: '/companies',
         }}
       />
 
-      <div className={styles.description}>
-        <p>{company.description}</p>
-      </div>
+      <ContentSection>
+        <ContentMain>
+          <i>
+            <DisplayContent content={company.description} />
+          </i>
+          <h3 className={styles.sectionHeader}>Kommende arrangementer</h3>
+          <ListEvents
+            events={upcomingEvents}
```

Comment:
This complains about the type of events. Haven't looked to much into it yet, so if anyone has any suggestions for a fix I'll gladly accept :innocent: 
 ```Type 'ListEvent[]' is not assignable to type 'Event[]'.ts(2322)```

Reply:
The `ListEvent` type used here (in `Props`) (imported from `app/store/models/event` is incorrect. Change the type to `Event` from `app/models` and it should be correct

Reply:
Great, thank you!